src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,205 @@
+                "Test failed! Expected " + WrongDataException.class.getName()
+                        + " Your age cannot be null!");
+    }
+
+    @Test
+    void register_youngAge_notOk() {
+        User user = new User(123456L, "First", "qwerty", 16);
+        assertThrows(WrongDataException.class, () -> registrationService.register(user),
+                "Test failed! Expected " + WrongDataException.class.getName()
+                        + "Your age cannot be less than " + 18 + " years!");
+    }
+
+    @Test
+    void register_oldAge_notOk() {
+        User user = new User(123456L, "Last", "qwerty", 101);
+        assertThrows(WrongDataException.class, () -> registrationService.register(user),
+                "Test failed! Expected " + WrongDataException.class.getName()
+                        + "Your age cannot be more than " + 100 + " years!");
+    }
+
+    @Test
+    void register_userExists_notOk() {
+        User user = new User(123456L, "First", "qwerty", 21);
+        storageDao.add(user);
+        assertThrows(WrongDataException.class, () -> registrationService.register(user),
+                "Test failed! Expected " + WrongDataException.class.getName()
+                        + " You cannot register user with the same login!");
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        User user = new User(123456L, "", "qwerty", 21);
+        assertThrows(WrongDataException.class, () -> registrationService.register(user),
+                "Test failed! Expected " + WrongDataException.class.getName()
+                        + " Login cannot be empty!");
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User(123456L, null, "qwerty", 21);
+        assertThrows(WrongDataException.class, () -> registrationService.register(user),
+                "Test failed! Expected " + WrongDataException.class.getName()
+                        + " Login cannot be null!");
+    }
+
+    @Test
+    void register_toLongLogin_notOk() {
+        User user = new User(123456L, "Polischuck", "qwerty", 21);
+        assertThrows(WrongDataException.class, () -> registrationService.register(user),
+                "Test failed! Expected " + WrongDataException.class.getName()
+                        + " Login must contain no more " + 6 + " characters");
+    }
+
+    @Test
+    void register_loginWithSpace_notOk() {
+        User user = new User(123456L, " Polis", "qwerty", 21);
+        assertThrows(WrongDataException.class, () -> registrationService.register(user),
+                "Test failed! Expected " + WrongDataException.class.getName()
+                        + " Login cannot contain spaces!");
+    }
+
+    @Test
+    void register_successRegister_ok() {
+        User user = new User(123456L, "Ford", "qwerty", 21);
+        User actual = registrationService.register(user);
+        assertEquals(user, actual, "Registration must be successful!");
```

Comment:
let's also check storage here
and also check if user has an id after registration (don't set id in constructor)